src/Microsoft.AspNet.Server.Kestrel/Http/Listener.cs

```diff
@@ -50,11 +50,11 @@ protected Listener(ServiceContext serviceContext) : base(serviceContext)
         /// </summary>
         protected abstract UvStreamHandle CreateListenSocket(string host, int port);
 
-        protected static void ConnectionCallback(UvStreamHandle stream, int status, Exception error, object state)
+        protected void ConnectionCallback(UvStreamHandle stream, int status, Exception error, object state)
         {
             if (error != null)
             {
-                Trace.WriteLine("Listener.ConnectionCallback " + error.ToString());
+                Log.LogError("Listener.ConnectionCallback ", error);
```

Comment:
Make this static again and change this to:

    ((Listener)state).Log.LogError("Listener.ConnectionCallback ", error);

You could make a local listener var and use it in the else block too if you want.